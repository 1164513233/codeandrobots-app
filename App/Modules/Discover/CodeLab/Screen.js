import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  InstructionList,
  CodeLabNav,
  Modal } from 'App/Components'

import { Metrics } from 'App/Themes'

import Types from 'App/Services/PropTypes'

export default class Screen extends Component {
  static propTypes = {
    config: PropTypes.any,
    instructions: PropTypes.arrayOf(PropTypes.shape({
      ...Types.instruction
    })).isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    onConnect: PropTypes.func.isRequired,
    onChangeOrder: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    onNavPress: PropTypes.func.isRequired,
    onRun: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)
    this.state = { navHeight: 0 }
  }

  onNavPress = (category, instruction) => {
    const { onNavPress } = this.props
    onNavPress(category, instruction)
  }

  onNavHeightChange = (navHeight) => {
    this.setState({ navHeight })
  }

  resolveSkills = () => {
    const { config } = this.props
    const { moves, skills } = config || {}

    const allSkills = []
    if (moves && moves.length > 0) {
      allSkills.push({
        category: 'Moves',
        items: moves
      })
    }
    if (skills && skills.length > 0) {
      allSkills.push(...skills)
    }

    return allSkills
  }

  render () {
    const {
      instructions,
      showNotConnectedModal,
      onConnect,
      onChangeOrder,
      onClose,
      onHideNotConnectedModal} = this.props
    const { navHeight } = this.state

    const skills = this.resolveSkills()

    return (
      <Container>
        <InstructionList
          style={{marginBottom: navHeight + Metrics.unit}}
          instructions={instructions}
          onChangeOrder={onChangeOrder}
          onClose={onClose} />
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onConnect}
          template='NotConnected' />
        <CodeLabNav
          skills={skills}
          onPress={this.onNavPress}
          onNavHeightChange={this.onNavHeightChange} />
      </Container>
    )
  }
}
