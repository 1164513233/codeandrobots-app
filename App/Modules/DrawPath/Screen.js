import React, { Component } from 'react'
import { Draw } from 'App/Components'

export default class DrawPathScreen extends Component {
  render () {
    return (
      <Draw strokeColor={'red'} strokeWidth={7} />
    )
  }
}
