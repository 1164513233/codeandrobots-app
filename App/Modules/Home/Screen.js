import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { isIOSProduction } from 'App/Services/Properties'

import { Container, List, ListItem, BottomNav } from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    onNavigatePress: PropTypes.func.isRequired
  }

  render () {
    const { onNavigatePress } = this.props
    return (
      <Container>
        {!isIOSProduction() &&
          <List>
            <ListItem
              title='Play & Explore'
              text='Let’s play'
              button='Play'
              onPress={() => { onNavigatePress('PlayerScreen') }} />
            <ListItem
              title='Code Lab'
              text='Let’s see what you can do'
              button='Code'
              onPress={() => { onNavigatePress('CodeLabScreen') }} />
            <ListItem
              title='Learn'
              text='You’ll be a guru in no time'
              button='Learn'
              onPress={() => { onNavigatePress('LearnScreen') }} />
            <ListItem
              title='Lab'
              text='Get under the hood'
              button='Lab'
              onPress={() => { onNavigatePress('LabScreen') }} />
          </List>
        }
        {isIOSProduction() &&
          <List>
            <ListItem
              title='Play & Explore'
              text='Let’s play'
              button='Play'
              onPress={() => { onNavigatePress('PlayScreen') }} />
            <ListItem
              title='Discover'
              text='Let’s see what you can do'
              button='Discover'
              onPress={() => { onNavigatePress('DiscoverScreen') }} />
            <ListItem
              title='Learn'
              text='You’ll be a guru in no time'
              button='Learn'
              onPress={() => { onNavigatePress('LearnScreen') }} />
          </List>
        }
        <BottomNav
          onRatePress={() => { onNavigatePress('RateScreen') }}
          onHomePress={() => { onNavigatePress('PlayerScreen') }}
          onSettingsPress={() => { onNavigatePress('SettingsScreen') }} />
      </Container>
    )
  }
}
