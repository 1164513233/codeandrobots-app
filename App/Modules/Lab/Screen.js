import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Links,
  Link,
  List,
  StatListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <List style={{ margin: Metrics.unit * 2 }} cols={2}>
          <StatListItem icon='bluetooth' title='bluetooth' value='Connected' />
          <StatListItem icon='battery' title='battery' value='85%' />
          <StatListItem icon='play' title='playtime' value='4.8h' />
          <StatListItem icon='code' title='remixes' value='5' />
          <StatListItem icon='book' title='lessons' value='2' />
          <StatListItem icon='gamepad' title='level' value='1' />
        </List>
        <Modal
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer>
          <Links>
            <Link text='Shop' onPress={onPress} />
            <Link text='Customize' onPress={onPress} />
            <Link text='Add a buddy' onPress={onPress} />
            <Link text='View logs' onPress={onPress} />
          </Links>
        </Footer>
      </Container>
    )
  }
}
