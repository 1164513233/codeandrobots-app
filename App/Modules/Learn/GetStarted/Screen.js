import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card } from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    onDone: PropTypes.func.isRequired,
    onLearnMorePress: PropTypes.func.isRequired
  }

  render () {
    const { onDone, onLearnMorePress } = this.props
    return (
      <Container>
        <Carousel onDone={onDone}>
          <Card image={Images.hello} title='Hi there ðŸ‘‹' />
          <Card
            image={Images.hello}
            text={'Sorry, the app is still under construction ðŸš§\n\nThe good news is you can follow along as this Beta version of the app develops over the coming weeks.'}
            button='Learn More'
            onPress={onLearnMorePress} />
        </Carousel>
      </Container>
    )
  }
}
