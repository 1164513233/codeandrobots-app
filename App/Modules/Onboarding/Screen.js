import React, { Component } from 'react'

import {
  Container,
  Carousel,
  Card
} from 'App/Components'

import { Colors, Metrics, Images, Videos } from 'App/Themes'

export default class Screen extends Component {

  render () {
    const { onDone } = this.props
    
    return (
      <Container>
        <Carousel
          buttons={['Next', 'Next', 'Done']}
          onPress={[null, null, null]}
          onDone={onDone}>
          <Card
            video={Videos.atlas}
            title='Code & Robots'
            text='Where your robots comes alive'
            style={{title_center: {fontSize: 128} }}/>
          <Card
            image={Images.hello}
            title='Robots await your commands'
            text='Otto & Nybble supported out of the box, more robots coming soon!'
          />
          <Card
            image={Images.done}
            title='Lots more to come! ðŸŽ‰'
            text='Learn to code, build your own robot and so much more...'
          />
        </Carousel>
      </Container>
    )
  }
}