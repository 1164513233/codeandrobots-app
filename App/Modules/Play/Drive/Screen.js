import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Joystick,
  Card} from 'App/Components'

import s from './Styles'

export default class Screen extends Component {
  static propTypes = {
    message: PropTypes.string.isRequired,
    onDraggableMove: PropTypes.func,
    onDraggableRelease: PropTypes.func,
    onDraggableStart: PropTypes.func
  }

  render () {
    const {
      message,
      onDraggableMove,
      onDraggableRelease,
      onDraggableStart
    } = this.props
    return (
      <Container>
        <View style={s.joystickView}>
          <Joystick
            onDraggableMove={onDraggableMove}
            onDraggableRelease={onDraggableRelease}
            onDraggableStart={onDraggableStart} />
        </View>
        <Footer>
          <Card text={message} />
        </Footer>
      </Container>
    )
  }
}
