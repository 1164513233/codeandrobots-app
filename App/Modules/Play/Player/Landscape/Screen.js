import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  PlayerCenterNav,
  PlayerSideParams,
  PlayerSideJoystick,
  Modal} from 'App/Components'

import s from './Styles'

export default class LandscapeScreen extends Component {
  static propTypes = {
    config: PropTypes.any,
    message: PropTypes.string.isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    onConnect: PropTypes.func,
    onDraggableMove: PropTypes.func,
    onDraggableRelease: PropTypes.func,
    onDraggableStart: PropTypes.func,
    onSliderPress: PropTypes.func.isRequired,
    onSkillPress: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired
  }

  render () {
    const {
      config,
      showNotConnectedModal,
      onConnect,
      onDraggableMove,
      onDraggableRelease,
      onDraggableStart,
      onSliderPress,
      onSkillPress,
      onHideNotConnectedModal
    } = this.props

    const { params } = config || {}

    // TODO Only one slider supported currently
    const slider = (config && params.length > 0)
      ? {...params[0], labels: params[0].values}
      : null

    return (
      <Container>
        <View style={s.view}>
          {slider &&
            <PlayerSideParams
              {...slider}
              onPress={onSliderPress} />
          }
          <PlayerCenterNav
            theme='primary'
            config={config}
            onSkillPress={onSkillPress} />
          <PlayerSideJoystick
            side='right'
            onDraggableMove={onDraggableMove}
            onDraggableRelease={onDraggableRelease}
            onDraggableStart={onDraggableStart} />
        </View>
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onConnect}
          template='NotConnected' />
      </Container>
    )
  }
}
