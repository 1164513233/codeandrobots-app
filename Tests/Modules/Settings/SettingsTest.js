import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Screen from 'App/Modules/Settings/Screen'

describe('Settings', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Screen
        onChooseRobotPress={() => {}}
        onShareAppPress={() => {}}
        onFeedbackPress={() => {}}
        onFAQPress={() => {}}
        onSupportPress={() => {}}
        onAboutPress={() => {}}
        onPrivacyPress={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
