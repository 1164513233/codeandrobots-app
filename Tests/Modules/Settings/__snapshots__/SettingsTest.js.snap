// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component renders correctly 1`] = `
<Container>
  <CompactListItem
    onPress={[Function]}
    title="Share app with a friend"
  />
  <CompactListItem
    onPress={[Function]}
    title="Share your ideas with us"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 256,
        "display": "flex",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Roboto-Medium",
          "fontSize": 14,
        }
      }
    >
      App Version N/A (N/A)
    </Text>
  </View>
  <Footer>
    <Links>
      <Link
        onPress={[Function]}
        text="FAQ"
      />
      <Link
        onPress={[Function]}
        text="Support"
      />
      <Link
        onPress={[Function]}
        text="About"
      />
      <Link
        onPress={[Function]}
        text="Privacy"
      />
    </Links>
  </Footer>
</Container>
`;
